sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -668,22 +681,35 @@ case class Greatest(children: Seq[Expression]) extends Expression {
 
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val evalChildren = children.map(_.genCode(ctx))
-    ctx.addMutableState(ctx.JAVA_BOOLEAN, ev.isNull)
-    ctx.addMutableState(ctx.javaType(dataType), ev.value)
-    def updateEval(eval: ExprCode): String = {
+    val isNull = ctx.freshName("greatestTmpIsNull")
```

Comment:
I think there is no need for it and we can use `ev.isNull`

Reply:
As I wrote in the description, we would like to make the lifetime of a global variable local.  
Sorry for missing addition of this comment.

Reply:
I am not sure about this. Because anyway, if `ev.isNull` is a global variable, here the problem would still exist since we are declaring it in the method (line 635). So I don't see how this changes/solves the problem.

Reply:
IIUC, at line 635 `ev.isNull` is declared as local variable.

Reply:
yes, thus if it is declared as global variable, we still have a global variable with the same name of a local one. Am I missing something?